src/Engine/EmitMSIL/CodeGenIL.cs

```diff
@@ -361,34 +363,19 @@ private Type EmitCoercionCode(AssociativeNode arg, Type argType, Type param)
             mInfo = typeof(IEnumerator).GetMethod(nameof(IEnumerator.MoveNext));
             EmitOpCode(OpCodes.Callvirt, mInfo);
 
-            var leaveLabel = DefineLabel();
             EmitOpCode(OpCodes.Brtrue_S, loopBodyLabel.Value);
-            EmitOpCode(OpCodes.Leave_S, leaveLabel.Value);
-
-            EmitOpCode(OpCodes.Ldloc, enumeratorIndex);
-
-            var finallyLabel = DefineLabel();
-            EmitOpCode(OpCodes.Brfalse_S, finallyLabel.Value);
 
             EmitOpCode(OpCodes.Ldloc, enumeratorIndex);
 
             mInfo = typeof(IDisposable).GetMethod(nameof(IDisposable.Dispose));
             EmitOpCode(OpCodes.Callvirt, mInfo);
 
-            MarkLabel(finallyLabel.Value, "label: finally label");
-            EmitOpCode(OpCodes.Endfinally);
-
-            MarkLabel(leaveLabel.Value, "label: exit label");
+           
             EmitOpCode(OpCodes.Ldloc, newArrIndex);
 
             localBuilder = DeclareLocal(t, "target array");
             var targetArrayIndex = localBuilder.LocalIndex;
             EmitOpCode(OpCodes.Stloc, targetArrayIndex);
-
-            var endLabel = DefineLabel();
-            EmitOpCode(OpCodes.Br_S, endLabel.Value);
-
-            MarkLabel(endLabel.Value, "label: end label");
```

Comment:
So all of this code removal is simply equivalent to removing the try and finally blocks?

Reply:
well, it removes the leave instruction, which if I am understanding correctly, jumped to the finally label before jumping to the leave label - it also removes the end label, we just fall into that code now instead of needing to branch unconditionally.